packages/hardhat-plugin/src/ui/components/execution/FinalStatus.tsx

```diff
@@ -0,0 +1,117 @@
+import { Box, Text } from "ink";
+import flattenDeep from "lodash.flattendeep";
+
+import { UiFuture, UiFutureStatusType, UiState } from "../../types";
+
+import { AddressResults } from "./AddressResults";
+import { Divider } from "./Divider";
+
+export const FinalStatus = ({ state }: { state: UiState }) => {
+  const allFutures = flattenDeep(state.batches);
```

Comment:
This can be done with a builtin array method now

```suggestion
  const allFutures = state.batches.flat(Infinity);
```

https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/array/flat

Reply:
Done